### PR TITLE
Added const `Postprocessor::getValue()`

### PR DESCRIPTION
**Pull request by joshuahansel**
This is step 1 of the transition plan for const `Postprocessor::getValue()`: #7770

### CODE REVIEW

**Comment by joshuahansel** [comment]
@loganharbour Any idea what's going on with the conda "Fetch and Branch" failures? Seems to be only on this PR.

**Comment by loganharbour** [comment]
> @loganharbour Any idea what's going on with the conda "Fetch and Branch" failures? Seems to be only on this PR.

Should be good now

**Comment by GiudGiud** [comment]
I can do the review

**Comment by joshuahansel** [comment]
@GiudGiud I was just about to ask for a brave volunteer. Thanks!

File: framework/src/postprocessors/Postprocessor.C [hunk 1]
```diff
@@ -52,3 +52,15 @@ Postprocessor::Postprocessor(const MooseObject * moose_object)
...
+PostprocessorValue
+Postprocessor::getValue()
+{
+  return static_cast<const Postprocessor *>(this)->getValue();
+}
+
+PostprocessorValue
+Postprocessor::getValue() const
+{
+  mooseError("getValue() (const or non-const) must be implemented.");
```
Comment (GiudGiud) [inline framework/src/postprocessors/Postprocessor.C:65]:
```suggestion
  mooseError("getValue() (const or non-const, const much preferrably) must be implemented.");
```

**Review by GiudGiud — APPROVED** [review]
all good

File: framework/src/postprocessors/Postprocessor.C (hunk 1 above)
Comment (GiudGiud) [reply framework/src/postprocessors/Postprocessor.C:65]: probably not needed if the whole fixing the herd does not drag on for 6 months

Comment (joshuahansel) [reply framework/src/postprocessors/Postprocessor.C:65]: Yeah, no one should hit this error message really.

**Comment by loganharbour** [comment]
Uh oh https://civet.inl.gov/job/1691468/

File: test/include/userobjects/SamplerTester.h [hunk 2]
```diff
@@ -31,7 +31,7 @@ class SamplerTester : public GeneralPostprocessor
   virtual void execute() final;
   virtual void initialize() final;
   virtual void finalize() final;
-  virtual Real getValue() final;
+  virtual Real getValue() const final;
```
Comment (GiudGiud) [inline test/include/userobjects/SamplerTester.h:34]: here

**Comment by GiudGiud** [comment]
I think this is just a missing `using Postprocessor::getValue` statement

**Comment by joshuahansel** [comment]
Guillaume is right - that's been a classic debug warning in this PR. I'm guessing that PP was added simultaneously. I'll put up the fix.

**Comment by joshuahansel** [comment]
Why is `SamplerTester` under `userobjects` when it's a PP?!

**Comment by joshuahansel** [comment]
> I'm guessing that PP was added simultaneously.

Apparently not. Looks 4 years old 😕

**Pull request merged**